orchestrator/workflows/modify_note.py

```diff
@@ -33,17 +33,33 @@ class ModifyNoteForm(SubmitFormPage):
         note: LongText = old_note
 
     user_input = yield ModifyNoteForm
-    subscription.note = user_input.note
-    db.session.add(subscription)
+
     return {
         "old_note": old_note,
         "note": user_input.note,
-        "subscription": to_serializable(subscription),
         "__old_subscriptions__": subscription_backup,
     }
 
 
+@step("Store note")
+def store_subscription_note(subscription_id: UUIDstr, note: str) -> State:
+    subscription = subscriptions.get_subscription(subscription_id)
+
+    subscription.note = note
+    db.session.add(subscription)
+
+    return {
+        "subscription": to_serializable(subscription),
+    }
+
+
 @workflow("Modify Note", initial_input_form=wrap_modify_initial_input_form(initial_input_form), target=Target.MODIFY)
 def modify_note() -> StepList:
     push_subscriptions = conditional(lambda _: app_settings.CACHE_DOMAIN_MODELS)
-    return init >> store_process_subscription(Target.MODIFY) >> push_subscriptions(cache_domain_models) >> done
+    return (
+        init
+        >> store_process_subscription(Target.MODIFY)
+        >> store_subscription_note
```

Comment:
Do we need to take some action (i.e. backfill migration) for historical modify_note workflows which do not have this step?

Reply:
They should serve as a log and in general it's a bad idea to change history. The workflow is so simple that they should all be in the state Done and thus can't be retried.

The past workflows without the `store_note` step don't show anything anomalous on the frontend.